### ISSUE TITLE
lpdd failing to download data

### ISSUE THREAD

**Issue by macbookandrew** [bachya/lp-vault-manager#11]
I ran across this workflow today and it looks wonderful—I’d love to use it. However, it doesn’t seem to be working correctly.

I got through the login process fine, but running `lpvs` returns no results. Running `lpdd` starts the process but it never finishes. If I quit the `lpass` process from Activity Monitor, the following data shows up in the debug log:

```
Starting debug for 'LastPass Vault Manager'

[ERROR: alfred.workflow.action.script] 16:18:11 workflow.py:1386 DEBUG    Reading settings from `/Users/andrew/Library/Application Support/Alfred 2/Workflow Data/com.bachya.lpvm/settings.json` ...
16:18:12 workflow.py:1960 DEBUG    Workflow version : 4.2.0
16:18:12 lpdd_exec.py:22 DEBUG    Exec arguments: [u'download-data']
16:18:12 lpdd_exec.py:34 DEBUG    Parsed command: download-data
16:18:12 lpdd_exec.py:35 DEBUG    Parsed argument: None
16:18:12 lpdd_exec.py:36 DEBUG    Parsed delimiter: >
16:18:12 lpdd_exec.py:43 DEBUG    Executing command: download-data
Please enter the LastPass master password for <myemailaddress@provider.com>.

Master Password: 16:18:24 utilities.py:70 ERROR    There was an issue with LastPass: Command '[u'/usr/local/bin/lpass', u'export']' returned non-zero exit status -15
16:18:24 workflow.py:1996 DEBUG    Workflow finished in 12.587 seconds.
```

Any idea what might be wrong? I did run `lpass` from the command line to enter my 2FA password and told it to trust the “browser,” so I don’t think that’s the issue. I just disabled 2FA on my LastPass account and am still seeing the same behavior.

Thanks

**Comment by Odaeus** [comment]
Same here, I'm logged in but the `lpass` command is requesting the master password again.

**Comment by bachya** [comment]
@macbookandrew: What happens if you run `/usr/local/bin/lpass export` from Terminal?

**Comment by macbookandrew** [comment]
It lists this message four times:
`pinentry-curses: no LC_CTYPE known - assuming UTF-8`
asks for my password, and upon submitting it, lists this info for apparently all my sites: URL, username, password, description.

See attached screenshot.
![screen-shot-2015-02-26-at-3 21 34-pm](https://cloud.githubusercontent.com/assets/784333/6401076/8de4df02-bdcb-11e4-88cd-20e458c8371c.jpg)

**Comment by kiantech** [comment]
Looks like the script hangs waiting for a password input (I think). I was able to get it to work by browsing to the workflows directory, and excute the python script manually 'python lpdd_exec.py download-data'. I entered my masterpassword and the export completed successfully.

**Comment by macbookandrew** [comment]
@kiantech Thanks—that tip worked for me. I’ll have to manually run it to refresh the cache as needed until there’s a fix for it.

**Comment by someinternetguy** [comment]
Hi, im having the same issue... I can update manually like @macbookandrew but doing so from alfred just hangs... would love for this to be fixes.  I love this workflow!!

**Comment by bachya** [comment]
Hi guys – appreciate your thoughts. Unfortunately, this seems to be associated with the monstrous #3 (which I cannot replicate for the life of me). Unless someone has the time to replicate the issue and dig into the code a bit more, I'm not sure how likely it'll be that I can fix. :(

**Comment by someinternetguy** [comment]
@bachya thanks for getting back.  I'm in no way a programmer so I can't do much to help fix the code.  However, if you or anyone needs debugging info or testing done, I can do that and send results.

**Comment by benbernard** [comment]
I dug into this a little bit, but haven't been able to come up with a working theory.  I am replicating the issue right now.  My current hypothesis is that it has to do with 2nd factor auth prompting? not sure about that.  Any of you other folks having problems have 2fa enabled?

**Comment by macbookandrew** [comment]
@benbernard Yes, I do have 2FA enabled. I tried disabling it, deleting the workflow, re-adding the workflow, logging in, and still it did not work.
